cdap-common/src/main/java/co/cask/cdap/common/conf/CConfigurationUtil.java

```diff
@@ -37,6 +40,21 @@ public static void copyTxProperties(CConfiguration cConf, org.apache.hadoop.conf
   }
 
   /**
+   * Get extra jars set in {@link CConfiguration} as a list of {@link File}.
+   *
+   * @param cConf {@link CConfiguration} containing the extra jars
+   * @return a list of {@link File} created from the extra jars set in cConf.
+   */
+  public static List<File> getExtraJars(CConfiguration cConf) {
+    String[] extraJars = cConf.getStrings(Constants.AppFabric.PROGRAM_CONTAINER_DIST_JARS);
+    List<File> jarFiles = new ArrayList<>();
+    for (String jarPath : extraJars) {
+      jarFiles.add(new File(jarPath));
```

Comment:
This actually implies the extra jars has to be files on the master node. I think it'll be better to support jars that are already on HDFS. This means this method should return `List<URI>` instead. For path that doesn't have protocol, we can assume it is local.